storage/interface.go

```diff
@@ -197,6 +197,9 @@ type SelectHints struct {
 	By       bool     // Indicate whether it is without or by.
 	Range    int64    // Range vector selector range in milliseconds.
 
+	ShardIndex uint64 // Current shard index (starts from 0 and up to ShardCount-1).
```

Comment:
Be nice to link to a higher-level description of the sharding concept.
E.g. the comment on `IndexReader` interface, or somewhere in the docs.

Reply:
Rather than linking, I've tried to better explain it directly here. Would this new comment work?
https://github.com/prometheus/prometheus/pull/10421/commits/e37b7f9576a6f200280d89f23ffdb6397458fdcb